site-service/src/main/java/uk/ac/ox/ndph/mts/site_service/service/SiteServiceImpl.java

```diff
@@ -76,4 +89,60 @@ public String save(Site site) {
         return sites;
     }
 
+    private ValidationResponse validateParentSiteExists(final String parentSiteId) {
+        if (this.siteStore.findById(parentSiteId).isPresent()) {
+            return new ValidationResponse(true, "");
+        }
+        return new ValidationResponse(false, Services.PARENT_NOT_FOUND.message());
+    }
+
+    /**
+     * Find Site By Site Name
+     *
+     * @param siteName the Site to search.
+     * @return site The Site being searched, or none() if not found
+     */
+    public Optional<Site> findSiteByName(String siteName) {
+        return siteStore.findByName(siteName);
+    }
+
+    /**
+     * Find site by ID
+     *
+     * @param id site ID to search for
+     * @return site if found
+     * @throws NotFoundException if not found
+     */
+    public Site findSiteById(String id) throws NotFoundException {
+        return this.siteStore
+                .findById(id)
+                .orElseThrow(() -> new NotFoundException(Services.SITE_NOT_FOUND.message(), id));
+
+    }
+
+    /**
+     * Test if the root node is present. If not, then the site service cannot be safely used,
+     * since the trial has not yet been initialized.
+     *
+     * @return true if a root site node is present
+     */
+    private boolean isRootSitePresent() {
+        return this.siteStore
+                .findRoot()
+                .isPresent();
+    }
+
+    /**
+     * Return the root node, throwing if not present
+     *
+     * @return site if found
+     * @throws NotFoundException if no root site (bad trial initialization)
+     */
+    public Site findRootSite() throws NotFoundException {
```

Comment:
I don't see this method used by the controller. Maybe keep it package-private for now?

Reply:
Done